### PR TITLE
4.x : Unit tests to create nodes in graph

### PR DESCRIPTION
**Pull request by deepakrkris**
Unit tests to create nodes in graph
    
   - [ ] Test the nodes created in workspace graph

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   "scripts": {
     "lint": "eslint .",
     "pretest": "node test/helpers/pretest.js",
-    "test": "npm run integration && npm run acceptance",
-    "unit": "mocha test/unit/*.js",
+    "test": "npm run unit && npm run integration && npm run acceptance",
+    "unit": "mocha test/unit/datamodel/*.js",
```
Comment (superkhau) [inline package.json]: mocha 'test/unit/**/*.js'

File: test/unit/datamodel/datasource.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const Workspace = require('../../../lib/workspace');
+
+describe('Graph : DataSource', function() {
+  let workspace;
+
+  beforeEach(function(done) {
+    workspace = new Workspace('/');
+    workspace.addDomain('DataSource');
+    done();
+  });
```
Comment (superkhau) [inline test/unit/datamodel/datasource.js]: http://loopback.io//doc/en/contrib/style-guide.html

File: test/unit/datamodel/datasource.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  describe('create datasource node', function() {
+    it('new node', function(done) {
+      new DataSource(workspace, 'test', {}, {});
+      expect(workspace.getNode('DataSource', 'test')).to.not.to.be.undefined();
```
Comment (superkhau) [inline test/unit/datamodel/datasource.js]: test for the right node properties

File: test/unit/datamodel/middleware.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+const Workspace = require('../../../lib/workspace');
+
+describe('Graph : Middleware', function() {
+  let workspace;
+
+  beforeEach(function(done) {
+    workspace = new Workspace('/');
+    workspace.addDomain('Middleware');
+    done();
+  });
```
Comment (superkhau) [inline test/unit/datamodel/middleware.js]: http://loopback.io//doc/en/contrib/style-guide.html

File: test/unit/datamodel/middleware.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  describe('create middleware node', function() {
+    it('new node', function(done) {
+      new Middleware(workspace, 'test', {}, {});
+      expect(workspace.getNode('Middleware', 'test')).to.not.to.be.undefined();
```
Comment (superkhau) [inline test/unit/datamodel/middleware.js]: ditto

File: test/unit/datamodel/model-method.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+describe('Graph : ModelMethod', function() {
+  let workspace, model, method;
+
+  beforeEach(function(done) {
+    workspace = new Workspace('/');
+    workspace.addDomain('ModelDefinition');
+    workspace.addDomain('ModelMethod');
+    model = new Model(workspace, 'test', {}, {});
+    done();
+  });
```
Comment (superkhau) [inline test/unit/datamodel/model-method.js]: http://loopback.io//doc/en/contrib/style-guide.html

File: test/unit/datamodel/model-method.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+  describe('create method node', function() {
+    it('create a method', function(done) {
```
Comment (superkhau) [inline test/unit/datamodel/model-method.js]: fyi, creates -- use imperative sentence like other loopback tests

File: test/unit/datamodel/model-method.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      method = new Method(workspace, 'testmethod', {}, {});
+      expect(workspace.getNode('ModelMethod', 'testmethod'))
+        .to.not.to.be.undefined();
```
Comment (superkhau) [inline test/unit/datamodel/model-method.js]: ditto

File: test/unit/datamodel/model-method.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      done();
+    });
+    it('add method to model', function(done) {
```
Comment (superkhau) [inline test/unit/datamodel/model-method.js]: space between its

File: test/unit/datamodel/model-method.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+      model.setMethod(method);
+      const methodArray = model.getMethodDefinitions();
+      expect(methodArray.length).to.be.equal(1);
```
Comment (superkhau) [inline test/unit/datamodel/model-method.js]: chai lengthOf -- http://chaijs.com/api/bdd/#method_length

File: test/unit/datamodel/datasource.js (hunk 3 above)
Comment (deepakrkris) [inline test/unit/datamodel/datasource.js]: there are no properties added

**Comment by deepakrkris** [comment]
@superkhau fixed

**Pull request merged**